error.go

```diff
@@ -34,11 +35,44 @@ import (
 // Errors which know their underlying cause should implement this interface to
 // be compatible with RootCause.
 //
-// We use an unexported "cause" method instead of "Cause" because we don't
-// want dig-internal causes to be confused with the cause of the user-provided
-// errors. (For example, if the users are using github.com/pkg/errors.)
+// We use unexported methods because we don't want dig-internal causes to be
+// confused with the cause of the user-provided errors. For example, if we
+// used Unwrap(), then user-provided methods would also be unwrapped by
+// RootCause. We want RootCause to eliminate the dig error chain only.
 type causer interface {
+	fmt.Formatter
+
+	// Returns the next error in the chain.
 	cause() error
+
+	// Writes the message or context for this error in the chain.
+	//
+	// verb is either %v or %+v.
+	writeMessage(w io.Writer, verb string)
```

Comment:
what if the verb assumption is violated? Should that be an explcit error case?

Reply:
We control invocations to writeMessage, so an unnecessary thing to guard against.

More generally, the contract is: `%+v` is special, `%v` is the default (per `fmt`), and everything else is undefined. Since you don't want string formatting to error out unnecessarily (you don't want the thing you're using for debugging to fail with a different error), we fall back to `%v` whenever it's not `%+v`.